src/Symfony/Component/PropertyAccess/PropertyAccessor.php

```diff
@@ -714,6 +714,19 @@ private function getWriteAccessInfo($class, $property, $value)
                     // we call the getter and hope the __call do the job
                     $access[self::ACCESS_TYPE] = self::ACCESS_TYPE_MAGIC;
                     $access[self::ACCESS_NAME] = $setter;
+                } elseif (null !== $this->findAdderAndRemover($reflClass, $singulars)) {
+                    $access[self::ACCESS_TYPE] = self::ACCESS_TYPE_NOT_FOUND;
+                    $access[self::ACCESS_NAME] = sprintf(
+                        'The property "%s" in class "%s" can be defined with the methods %s but '.
+                        'the new value must be an array or an instance of \Traversable, '.
+                        '"%s" given.',
+                        $property,
+                        $reflClass->name,
+                        implode(' and ', array_map(function ($method) {
```

Comment:
no need for array_map here. "and" is also going to be repeated many times, let's replace it with a comma:
`implode('()", "', $methods)` and use `methods "%s()" but` in the message above.